### PR TITLE
Fix: Detect ternary operator in operator-linebreak rule

### LINKED ISSUE #3274: Detect ternary operator in operator-linebreak rule
For example, `after` disallows:

```
foo = 1
    + 2;
```

but

```
foo = true
    ? 1
    : 2;
```

generates no warnings. Should this rule also match a `ConditionalExpression`?

### PR DESCRIPTION
**Pull request by BYK**
Fixes #3274

### CODE REVIEW

**Comment by BYK** [comment]
If this looks like a good direction, I'll add more tests and update the docs.

**Comment by nzakas** [comment]
Looks fine to me.

**Comment by BYK** [comment]
Ready for review.

File: lib/rules/operator-linebreak.js [hunk 1]
```diff
@@ -92,14 +114,33 @@ module.exports = function(context) {
...
+        "enum": ["after", "before", "none", null]
+    },
+    {
+        "type": "object",
+        "properties": {
+            "overrides": {
+                "type": "object",
+                "properties": {
+                    "anyOf": {
+                        "type": "string",
```
Comment (BYK) [inline lib/rules/operator-linebreak.js]: Ideally we would also enforce a valid operator as the property name but I couldn't find an easy way to do that with JSON-schema.

**Comment by nzakas** [comment]
Can you squash your commits?

**Comment by BYK** [comment]
Oops, sorry. I forget to `--amend` sometimes :(

**Comment by nzakas** [comment]
LGTM

**Pull request merged**